addons/website_sale/tests/test_website_sale_invoice.py

```diff
@@ -33,3 +33,58 @@ def test_automatic_invoice_website_id(self):
 
         self.assertEqual(self.sale_order.website_id.id, self.website.id)
         self.assertEqual(self.sale_order.invoice_ids.website_id.id, self.website.id)
+
+    def test_automatic_invoice_author_uses_invoice_user_id(self):
+        """Test that automatic invoice email author uses invoice_user_id instead of portal user"""
+        # 1. Enable automatic invoicing
+        self.env['ir.config_parameter'].sudo().set_param('sale.automatic_invoice', 'True')
+
+        # 2. Create portal user (simulating website login)
+        portal_user = self.env['res.users'].create({
+            'name': 'Portal Customer',
+            'login': 'portal@test.com',
+            'email': 'portal@test.com',
+            'partner_id': self.partner_a.id,
+            'groups_id': [(6, 0, [self.env.ref('base.group_portal').id])],
+        })
+
+        # 3. Create salesperson
+        salesperson = self.env['res.users'].create({
+            'name': 'Sales Person',
+            'login': 'sales@test.com',
+            'email': 'sales@test.com',
+            'groups_id': [(6, 0, [self.env.ref('sales_team.group_sale_salesman').id])],
+        })
+
+        # 4. Create sale order as portal user (simulating website cart/checkout)
```

Comment:
Wrong comment, ecommerce orders are created with the SUPERUSER (see `sale_get_order`)